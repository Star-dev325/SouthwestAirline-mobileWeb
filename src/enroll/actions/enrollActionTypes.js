import { actionCreator } from 'src/shared/redux/actionCreator';

const { createTypes, createApiActions } = actionCreator('enroll');

const types = {
  sync: [],
  async: ['FETCH_SECURITY_QUESTIONS', 'CREATE_ACCOUNT']
};

export const apiActionCreator = createApiActions;
export default createTypes(types);
