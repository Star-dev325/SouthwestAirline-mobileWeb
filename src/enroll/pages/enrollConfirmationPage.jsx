// @flow
import React, { Component } from 'react';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { connect } from 'react-redux';
import _ from 'lodash';
import { ENROLL_PERSONAL_INFO_FORM } from 'src/shared/constants/formIds';
import { buildPathWithParamAndQuery } from 'src/shared/helpers/pathUtils';
import Container from 'src/shared/components/container';
import Message from 'src/shared/components/message';
import Icon from 'src/shared/components/icon';
import SubHeader from 'src/shared/components/subHeader';
import i18n from '@swa-ui/locale';

import type { Push } from 'src/shared/flow-typed/shared.types';

type PersonalInfoType = {
  firstName: string,
  lastName: string
};

type Props = {
  personalInfo: PersonalInfoType,
  accountNumber: string,
  push: Push
};

export class EnrollConfirmationPage extends Component<Props> {
  _onClickLoginButton = (evt: Event) => {
    evt.preventDefault();
    this.props.push(buildPathWithParamAndQuery('/login', null, { to: '/', simpleLogin: true, withPoints: false }));
  };

  render() {
    const {
      personalInfo: { firstName, lastName },
      accountNumber
    } = this.props;
    const fullName = [firstName, lastName].join(' ');

    return (
      <div className="enroll-confirmation-page">
        <div className="enroll-confirmation__bg">
          <SubHeader title="Confirmation" />
          <Container>
            <Message status="success" size="huge">
              {i18n('ENROLL_WELCOME')}, {firstName}!
            </Message>

            <p className="congratulation-text">
              {i18n('ENROLL_CONGRATULATION_TEXT_1')}
              <sup>Â®</sup>
              {i18n('ENROLL_CONGRATULATION_TEXT_2')}
            </p>

            <div className="confirmation-cards-page--wrapper">
              <div className="enroll-confirmation-page--cards">
                <img
                  src="/content/mkt/images/landing_pages/swa_logo_light.svg"
                  className="enroll-confirmation-page--cards-logo"
                />
                <span className="enroll-confirmation-page--cards-tm">{i18n('ENROLL_RAPID_REWARDS')}</span>
                <div className="enroll-confirmation-page--cards-user-info">
                  <div className="username">{fullName}</div>
                  <div className="account-number">{accountNumber}</div>
                </div>
              </div>
            </div>
            <div className="goto-login" onClick={this._onClickLoginButton}>
              <span>{i18n('ENROLL_LOG_IN')}</span>
              <Icon type="keyboard-arrow-right" />
            </div>
          </Container>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  personalInfo: {
    firstName: _.get(state, `app.formData.${ENROLL_PERSONAL_INFO_FORM}.data.firstName`),
    lastName: _.get(state, `app.formData.${ENROLL_PERSONAL_INFO_FORM}.data.lastName`)
  },
  accountNumber: _.get(state, 'app.account.accountNumber')
});

export default _.flowRight(withConnectedReactRouter, connect(mapStateToProps, {}))(EnrollConfirmationPage);
