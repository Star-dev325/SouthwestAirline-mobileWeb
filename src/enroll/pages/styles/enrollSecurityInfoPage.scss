.enroll-security-info-form {
  @include m0;

  .security-question-nav-field {
    @include mt5;
  }

  .fields {
    @include large;
  }

  .fields--label {
    @include bold;
    @include large;
    @include gray5;
    @include mb5;
  }

  .input {
    @include regular;
    @include mt4;
  }

  .checkbox-button {
    @include mt4;
  }

  .helper-text {
    @include gray5;
    @include medium;

    .helper-text--title {
      @include my4;
    }

    .helper-text--special-character-specification {
      @include ml6;
      @include mt4;
    }

    .helper-text--special-character_include {
      @include pdkblue;
      @include bold;
    }

    .helper-text--special-character_not-include {
      @include ml6;
    }

    .helper-text--special-character_hint {
      @include mt4;
    }
  }

  .segment {
    @include bggray2;
    @include mb4;

    &.secondary.fill {
      @include p0;
    }

    &.blue {
      @include bgpblue;
      padding: 1.6rem 1.2rem;
    }
  }
}
