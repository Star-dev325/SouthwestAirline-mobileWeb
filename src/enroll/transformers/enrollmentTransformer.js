import _ from 'lodash';
import { removeSeparator } from 'src/shared/helpers/separatorHelper';
import countryCodes from 'src/shared/constants/countryCode';

const DEFAULT_ADDRESS_PHONE_TYPE = 'HOME';
const DEFAULT_ISO_COUNTRY_CODE = 'US';
const DEFAULT_PHONE_COUNTRY_CODE = '1';

const getPhoneCountryCodeFromCountryCode = (countryCode) =>
  _.get(countryCodes, `${countryCode}`, DEFAULT_PHONE_COUNTRY_CODE).toString();

export const transformToEnrollRequest = (data) => {
  const {
    personalInfoData,
    contactInfoData: { optInForEmailSubscriptions = true, ...restContactInfoData } = {},
    securityInfoData
  } = data;

  return {
    userName: _.trim(securityInfoData.userName),
    password: securityInfoData.password,
    promoCode: _.isEmpty(securityInfoData.promoCode) ? '' : securityInfoData.promoCode,
    contactInfo: {
      address: _.merge(
        {},
        _.pick(restContactInfoData, ['addressLine1', 'addressLine2', 'city', 'stateProvinceRegion', 'zipOrPostalCode']),
        {
          isoCountryCode: _.isEmpty(restContactInfoData.isoCountryCode)
            ? DEFAULT_ISO_COUNTRY_CODE
            : restContactInfoData.isoCountryCode,
          addressType: DEFAULT_ADDRESS_PHONE_TYPE
        }
      ),
      phone: {
        countryCode: getPhoneCountryCodeFromCountryCode(restContactInfoData.phoneCountryCode),
        number: removeSeparator(restContactInfoData.phoneNumber),
        phoneType: DEFAULT_ADDRESS_PHONE_TYPE
      },
      emailAddress: restContactInfoData.email
    },
    customerInfo: {
      name: {
        firstName: _.trim(personalInfoData.firstName),
        lastName: _.trim(personalInfoData.lastName),
        middleName: _.trim(personalInfoData.middleName),
        preferredName: _.trim(personalInfoData.preferredName),
        // missing field will broken api
        suffix: _.isEmpty(personalInfoData.suffix) ? undefined : personalInfoData.suffix
      },
      gender: personalInfoData.gender,
      birthDate: personalInfoData.dateOfBirth
    },
    optInForEmailSubscriptions,
    securityQuestions: [
      {
        question: securityInfoData.question1,
        answer: securityInfoData.answer1
      },
      {
        question: securityInfoData.question2,
        answer: securityInfoData.answer2
      }
    ]
  };
};
