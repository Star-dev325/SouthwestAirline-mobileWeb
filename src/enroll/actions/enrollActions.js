// @flow
import { push } from 'connected-react-router';
import enrollActionTypes, { apiActionCreator } from 'src/enroll/actions/enrollActionTypes';
import { saveAccountNumber } from 'src/shared/actions/accountActions';
import FlowStatusActions from 'src/shared/actions/flowStatusActions';
import * as AccountsApi from 'src/shared/api/accountsApi';
import { STATUS } from 'src/shared/constants/flowConstants';
import { playHapticFeedback } from 'src/shared/helpers/hapticFeedbackHelper';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';

import type { Dispatch as ReduxDispatch } from 'redux';
import type { CreateUserAccountType } from 'src/shared/flow-typed/shared.types';

const { ENROLL__CREATE_ACCOUNT, ENROLL__FETCH_SECURITY_QUESTIONS } = enrollActionTypes;

const { fetchSecurityQuestions, fetchSecurityQuestionsFailed, fetchSecurityQuestionsSuccess } = apiActionCreator(
  ENROLL__FETCH_SECURITY_QUESTIONS
);

export const getSecurityQuestions = () => (dispatch: ReduxDispatch<*>) => {
  dispatch(fetchSecurityQuestions());

  return AccountsApi.getSecurityQuestions()
    .then((res) => dispatch(fetchSecurityQuestionsSuccess(res?.securityQuestions ?? [])))
    .catch((error) => dispatch(fetchSecurityQuestionsFailed(error)));
};

const { createAccount, createAccountSuccess, createAccountFailed } = apiActionCreator(ENROLL__CREATE_ACCOUNT);

export const createUserAccount = (request: CreateUserAccountType) => (dispatch: ReduxDispatch<*>) => {
  dispatch(createAccount(request));

  return AccountsApi.createAccount(request)
    .then((res) => {
      dispatch(saveAccountNumber(res?.accountNumber));
      dispatch(createAccountSuccess(res));
      dispatch(FlowStatusActions.setFlowStatus('enroll', STATUS.COMPLETED));
      dispatch(push(getNormalizedRoute({ routeName: 'confirmation' })));
      playHapticFeedback();
    })
    .catch((error) => dispatch(createAccountFailed(error)));
};
