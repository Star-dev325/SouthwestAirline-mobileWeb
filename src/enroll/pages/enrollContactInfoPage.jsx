// @flow
import _ from 'lodash';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import EnrollContactInfoForm from 'src/enroll/components/enrollContactInfoForm';
import * as AnalyticsActions from 'src/shared/analytics/actions/analyticsActions';
import ProgressionBar from 'src/shared/components/progressionBar';
import { ENROLL_CONTACT_INFO_FORM } from 'src/shared/constants/formIds';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';

import type { Push } from 'src/shared/flow-typed/shared.types';

type Props = {
  analyticsTrackSubmitFormFn: (string) => void,
  push: Push
};

export class EnrollContactInfoPage extends Component<Props> {
  _onSubmit = () => {
    const { push, analyticsTrackSubmitFormFn } = this.props;

    analyticsTrackSubmitFormFn('enroll-contact-info');
    push(getNormalizedRoute({ routeName: 'security' }));
  };

  render() {
    return (
      <div>
        <ProgressionBar
          currentIconType="user"
          step={2}
          subTitles={['Personal', 'Contact', 'Security']}
          title="Contact Info"
          totalStep={3}
        />
        <EnrollContactInfoForm formId={ENROLL_CONTACT_INFO_FORM} onSubmit={this._onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = {
  analyticsTrackSubmitFormFn: AnalyticsActions.trackSubmitForm
};

export default _.flowRight(withConnectedReactRouter, connect(null, mapDispatchToProps))(EnrollContactInfoPage);
