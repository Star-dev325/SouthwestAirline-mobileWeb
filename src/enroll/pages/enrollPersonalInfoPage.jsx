// @flow
import _ from 'lodash';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import EnrollPersonalInfoForm from 'src/enroll/components/enrollPersonalInfoForm';
import FlowStatusActions from 'src/shared/actions/flowStatusActions';
import * as AnalyticsActions from 'src/shared/analytics/actions/analyticsActions';
import ProgressionBar from 'src/shared/components/progressionBar';
import { STATUS } from 'src/shared/constants/flowConstants';
import { ENROLL_PERSONAL_INFO_FORM } from 'src/shared/constants/formIds';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { getNormalizedRoute } from 'src/shared/helpers/urlHelper';
import { store } from 'src/shared/redux/createStore';

import type { Push } from 'src/shared/flow-typed/shared.types';

type Props = {
  analyticsTrackSubmitFormFn: (string) => void,
  fieldNameEnabledForChange?: string,
  LOYALTY_AGE_VERIFICATION: boolean,
  push: Push
};

export class EnrollPersonalInfoPage extends Component<Props> {
  _onSubmit = () => {
    const { push, analyticsTrackSubmitFormFn } = this.props;

    store.dispatch(FlowStatusActions.setFlowStatus('enroll', STATUS.IN_PROGRESS));
    analyticsTrackSubmitFormFn('enroll-personal-info');
    push(getNormalizedRoute({ routeName: 'contact' }));
  };

  render() {
    const { LOYALTY_AGE_VERIFICATION, fieldNameEnabledForChange } = this.props;

    return (
      <div>
        <ProgressionBar
          currentIconType="user"
          step={1}
          subTitles={['Personal', 'Contact', 'Security']}
          title="Personal Info"
          totalStep={3}
        />
        <EnrollPersonalInfoForm
          fieldNameEnabledForChange={fieldNameEnabledForChange}
          formId={ENROLL_PERSONAL_INFO_FORM}
          LOYALTY_AGE_VERIFICATION={LOYALTY_AGE_VERIFICATION}
          onSubmit={this._onSubmit}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  fieldNameEnabledForChange: state?.app?.formData?.ENROLL_PERSONAL_INFO_FORM?.fieldNameEnabledForChange,
  LOYALTY_AGE_VERIFICATION: state?.app?.toggles?.LOYALTY_AGE_VERIFICATION
});

const mapDispatchToProps = {
  analyticsTrackSubmitFormFn: AnalyticsActions.trackSubmitForm
};

export default _.flowRight(
  withConnectedReactRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(EnrollPersonalInfoPage);
