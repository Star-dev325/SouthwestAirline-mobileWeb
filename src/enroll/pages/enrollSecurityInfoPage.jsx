// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import ProgressionBar from 'src/shared/components/progressionBar';
import * as EnrollActions from 'src/enroll/actions/enrollActions';
import * as AnalyticsActions from 'src/shared/analytics/actions/analyticsActions';
import * as FormDataActions from 'src/shared/actions/formDataActions';
import EnrollSecurityInfoForm from 'src/enroll/components/enrollSecurityInfoForm';
import { transformToEnrollRequest } from 'src/enroll/transformers/enrollmentTransformer';
import {
  ENROLL_CONTACT_INFO_FORM,
  ENROLL_PERSONAL_INFO_FORM,
  ENROLL_SECURITY_INFO_FORM
} from 'src/shared/constants/formIds';

import type { CreateUserAccountType } from 'src/shared/flow-typed/shared.types';
import type { FormData } from 'src/shared/form/flow-typed/form.types';

type Props = {
  minorAgeThreshold: number,
  dateOfBirth: string,
  enrollSecurityQuestions: Array<string>,
  getEnrollSecurityQuestionsFn: () => void,
  createUserAccountFn: (request: CreateUserAccountType) => void,
  personalInfoData: FormData,
  contactInfoData: FormData,
  securityInfoData: FormData,
  analyticsTrackSubmitFormFn: (string) => void,
  updateFormDataValueFn: (formId: string, fieldValues: *) => void
};

export class EnrollSecurityInfoPage extends Component<Props> {
  componentDidMount() {
    const { getEnrollSecurityQuestionsFn } = this.props;

    getEnrollSecurityQuestionsFn();
  }

  _onSubmit = () => {
    const { personalInfoData, contactInfoData, securityInfoData, createUserAccountFn, analyticsTrackSubmitFormFn } =
      this.props;
    const request = transformToEnrollRequest({ personalInfoData, contactInfoData, securityInfoData });

    createUserAccountFn(request);
    analyticsTrackSubmitFormFn('enroll-security-info');
  };

  _clearSecurityFields = () => {
    const { updateFormDataValueFn } = this.props;

    updateFormDataValueFn(ENROLL_SECURITY_INFO_FORM, {
      password: '',
      confirmedPassword: '',
      answer1: '',
      answer2: ''
    });
  };

  render() {
    const { enrollSecurityQuestions, dateOfBirth, minorAgeThreshold } = this.props;

    return (
      <div>
        <ProgressionBar
          step={3}
          totalStep={3}
          title="Security Info"
          subTitles={['Personal', 'Contact', 'Security']}
          currentIconType="user"
        />
        <EnrollSecurityInfoForm
          formId={ENROLL_SECURITY_INFO_FORM}
          securityQuestions={enrollSecurityQuestions}
          minorAgeThreshold={minorAgeThreshold}
          dateOfBirth={dateOfBirth}
          onSubmit={this._onSubmit}
          onValidationFailed={this._clearSecurityFields}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  minorAgeThreshold: _.get(state, 'app.wcmContent.applicationProperties.coppa.rrEnrollment.minAgeThreshold', 13),
  dateOfBirth: _.get(state, `app.formData.${ENROLL_PERSONAL_INFO_FORM}.data.dateOfBirth`),
  enrollSecurityQuestions: _.get(state, 'app.enroll.securityQuestions'),
  personalInfoData: _.get(state, `app.formData.${ENROLL_PERSONAL_INFO_FORM}.data`),
  contactInfoData: _.get(state, `app.formData.${ENROLL_CONTACT_INFO_FORM}.data`),
  securityInfoData: _.get(state, `app.formData.${ENROLL_SECURITY_INFO_FORM}.data`)
});

const mapDispatchToProps = {
  getEnrollSecurityQuestionsFn: EnrollActions.getSecurityQuestions,
  createUserAccountFn: EnrollActions.createUserAccount,
  analyticsTrackSubmitFormFn: AnalyticsActions.trackSubmitForm,
  updateFormDataValueFn: FormDataActions.updateFormDataValue
};

export default _.flowRight(
  withConnectedReactRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(EnrollSecurityInfoPage);
