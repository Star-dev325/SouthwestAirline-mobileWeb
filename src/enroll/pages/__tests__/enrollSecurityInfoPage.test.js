import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { EnrollSecurityInfoPage } from 'src/enroll/pages/enrollSecurityInfoPage';
import * as EnrollmentTransformer from 'src/enroll/transformers/enrollmentTransformer';
import * as AnalyticsActions from 'src/shared/analytics/actions/analyticsActions';
import EnrollSecurityQuestionsBuilder from 'test/builders/model/enrollSecurityQuestionsBuilder';
import { createMockedFormStore } from 'test/unit/helpers/formTestUtils';

describe('EnrollSecurityInfoPage', () => {
  const requestObj = {
    contactInfo: {
      address: { addressType: 'HOME', isoCountryCode: 'US' },
      emailAddress: undefined,
      phone: { countryCode: '1', number: undefined, phoneType: 'HOME' }
    },
    customerInfo: {
      birthDate: undefined,
      gender: undefined,
      name: {
        firstName: '',
        lastName: '',
        middleName: '',
        preferredName: '',
        suffix: undefined
      }
    },
    optInForEmailSubscriptions: true,
    password: undefined,
    promoCode: '',
    securityQuestions: [
      { answer: undefined, question: undefined },
      { answer: undefined, question: undefined }
    ],
    userName: ''
  };
  const createUserAccountFnMock = jest.fn();
  const getEnrollSecurityQuestionsFnMock = jest.fn();
  const analyticsTrackSubmitFormFnMock = jest.spyOn(AnalyticsActions, 'trackSubmitForm');
  const updateFormDataValueFnMock = jest.fn();
  const transformToEnrollRequestMock = jest
    .spyOn(EnrollmentTransformer, 'transformToEnrollRequest')
    .mockReturnValue(requestObj);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    const { container } = createComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render progress bar correctly', () => {
    const { container } = createComponent();

    expect(container.querySelector('.step-item--inner')).not.toBeNull();
  });

  it('should render enroll security info form', () => {
    const { container } = createComponent();

    expect(container.querySelector('.enroll-security-info-form')).not.toBeNull();
  });

  it('calls get security questions on mount', () => {
    createComponent();

    expect(getEnrollSecurityQuestionsFnMock).toHaveBeenCalled();
  });

  describe('submit form', () => {
    it('should call push and analytics track form when form is successfully submitted', () => {
      const instance = React.createRef();

      createComponent({ ref: instance });

      const mockData = {
        userName: 'testacct123',
        password: 'Test1234',
        confirmedPassword: 'Test1234',
        question1: 'some question',
        answer1: 'some answer',
        question2: 'another question',
        answer2: 'another answer'
      };

      instance.current._onSubmit(mockData);

      expect(analyticsTrackSubmitFormFnMock).toHaveBeenCalledWith('enroll-security-info');
      expect(createUserAccountFnMock).toHaveBeenCalledWith(requestObj);
      expect(updateFormDataValueFnMock).not.toHaveBeenCalled();
      expect(transformToEnrollRequestMock).toHaveBeenCalled();
    });

    it('should trigger form validation and display error header when Continue button is pressed and required fields are empty', () => {
      const { container } = createComponent();

      fireEvent.click(container.querySelector('.checkbox-button'));

      fireEvent.submit(container.querySelector('.enroll-security-info-form'));

      expect(container.querySelectorAll('div.error-header')[0].textContent).toEqual(
        'Please correct the highlighted errors.'
      );
      expect(analyticsTrackSubmitFormFnMock).not.toHaveBeenCalled();
      expect(createUserAccountFnMock).not.toHaveBeenCalled();
      expect(updateFormDataValueFnMock).toHaveBeenCalledWith('ENROLL_SECURITY_INFO_FORM', {
        password: '',
        confirmedPassword: '',
        answer1: '',
        answer2: ''
      });
      expect(transformToEnrollRequestMock).not.toHaveBeenCalled();
    });
  });

  const minorAcknowledge = 'I acknowledge that this enrollment is for a Customer who is under 13 years old.';
  const rulesAcknowledge = 'I acknowledge I have read and accept the rules';
  const minorAgeThreshold = 13;
  const enrollSecurityQuestions = new EnrollSecurityQuestionsBuilder().build();

  const createComponent = (props = {}) => {
    const defaultProps = {
      minorAcknowledge,
      rulesAcknowledge,
      minorAgeThreshold,
      dateOfBirth: '2000-01-23',
      enrollSecurityQuestions,
      getEnrollSecurityQuestionsFn: getEnrollSecurityQuestionsFnMock,
      createUserAccountFn: createUserAccountFnMock,
      personalInfoData: {},
      contactInfoData: {},
      securityInfoData: {},
      analyticsTrackSubmitFormFn: analyticsTrackSubmitFormFnMock,
      updateFormDataValueFn: updateFormDataValueFnMock
    };
    const state = {
      app: {},
      router: {
        location: {
          search: '_modal=SECURITY_QUESTION1_LIST_MODAL_ID'
        }
      }
    };

    const finalProps = { ...defaultProps, ...props };

    return render(
      <Provider store={createMockedFormStore(state)}>
        <EnrollSecurityInfoPage {...finalProps} />
      </Provider>
    );
  };
});
