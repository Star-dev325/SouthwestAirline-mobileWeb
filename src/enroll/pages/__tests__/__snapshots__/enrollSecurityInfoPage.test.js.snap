// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnrollSecurityInfoPage should render 1`] = `
<div>
  <div>
    <div>
      <div
        class="page-header page-header--no-padding"
        id=""
      >
        <div
          class="progression-bar"
        >
          <div
            class="title"
          >
            Security Info
          </div>
          <div
            class="progress-step"
          >
            <div
              class="progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="100"
                class="progress-bar"
                role="progressbar"
                style="width: 100%;"
              />
            </div>
            <span
              class="step-item step-item--completed"
              style="left: 0%;"
            >
              <span
                class="step-item--inner"
              >
                <i
                  class="completed-check icon icon_check"
                />
              </span>
            </span>
            <span
              class="step-item step-item--completed"
              style="left: 50%;"
            >
              <span
                class="step-item--inner"
              >
                <i
                  class="completed-check icon icon_check"
                />
              </span>
            </span>
            <span
              class="step-item step-item--current"
              style="left: 100%;"
            >
              <span
                class="step-item--inner"
              >
                <i
                  class="icon icon_user"
                />
              </span>
            </span>
          </div>
          <div
            class="progression-bar--sub-title-flex"
          >
            <div
              class="progression-bar--sub-title"
            >
              Personal
            </div>
            <div
              class="progression-bar--sub-title"
            >
              Contact
            </div>
            <div
              class="progression-bar--sub-title"
            >
              Security
            </div>
          </div>
        </div>
      </div>
      <div />
    </div>
    <div>
      <form
        class="enroll-security-info-form form"
        name="ENROLL_SECURITY_INFO_FORM"
        novalidate=""
      >
        <fieldset>
          <div
            class="segments"
          >
            <div
              class="secondary segment"
            >
              <div
                class="fields divided grouped"
              >
                <label
                  class="fields--label"
                >
                  ACCOUNT ACCESS
                </label>
                <div
                  class="field"
                >
                  <div
                    class="input"
                  >
                    <input
                      class=""
                      name="userName"
                      placeholder="Username"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="input"
                  >
                    <input
                      class=""
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="input"
                  >
                    <input
                      class=""
                      name="confirmedPassword"
                      placeholder="Confirm password"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="helper-text"
                >
                  <p
                    class="helper-text--title"
                  >
                    ENROLL_HELPER_TEXT_PASSWORD_TITLE
                  </p>
                  <p
                    class="helper-text--uppercase-letter"
                  >
                    ENROLL_HELPER_TEXT_PASSWORD_RULE_UPPERCASE_LETTER
                  </p>
                  <p
                    class="helper-text--number"
                  >
                    ENROLL_HELPER_TEXT_PASSWORD_RULE_NUMBER
                  </p>
                  <div>
                    ENROLL_HELPER_TEXT_PASSWORD_RULE_SPECIAL_CHARACTER
                    <div
                      class="helper-text--special-character-specification"
                    >
                      <span
                        class="helper-text--special-character_include"
                      >
                        ENROLL_HELPER_TEXT_PASSWORD_SPECIAL_CHARACTERS
                      </span>
                      <span
                        class="helper-text--special-character_not-include"
                      >
                        ENROLL_HELPER_TEXT_PASSWORD_SPECIAL_CHARACTERS_NOT_INCLUDE
                      </span>
                      <p
                        class="helper-text--special-character_hint"
                      >
                        ENROLL_HELPER_TEXT_PASSWORD_SPECIAL_CHARACTERS_HINT
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="secondary segment"
            >
              <div
                class="fields divided grouped"
              >
                <label
                  class="fields--label"
                >
                  SECURITY QUESTIONS
                </label>
                <div
                  class="field"
                >
                  <a
                    class="nav-item-link nav-item-field security-question-nav-field"
                    name="question1"
                  >
                    <div
                      class="fullwidth"
                    >
                      <span>
                        Choose your security question
                      </span>
                      <div
                        class="nav-item-field-value"
                        data-qa="nav-item-field-value"
                      />
                    </div>
                    <i
                      class="nav-item-link--icon icon icon_keyboard-arrow-right"
                    />
                  </a>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="input"
                  >
                    <input
                      class=""
                      name="answer1"
                      placeholder="Answer"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="field"
                >
                  <a
                    class="nav-item-link nav-item-field security-question-nav-field"
                    name="question2"
                  >
                    <div
                      class="fullwidth"
                    >
                      <span>
                        Choose your security question
                      </span>
                      <div
                        class="nav-item-field-value"
                        data-qa="nav-item-field-value"
                      />
                    </div>
                    <i
                      class="nav-item-link--icon icon icon_keyboard-arrow-right"
                    />
                  </a>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="input"
                  >
                    <input
                      class=""
                      name="answer2"
                      placeholder="Answer"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="field"
                >
                  <div
                    class="input"
                  >
                    <input
                      class=""
                      name="promoCode"
                      placeholder="Enrollment promo code (optional)"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="field"
              >
                <div
                  class="checkbox-button relative flex flex-cross-start"
                  data-qa="acceptRulesCheckbox"
                  name="acceptRulesAndRegulations"
                >
                  <div>
                    <span
                      class="checkbox-button--mark"
                    >
                      <i
                        class="absolute xlarge bgtransp icon icon_check"
                      />
                    </span>
                  </div>
                  <div>
                    <div
                      class="checkbox-button--children flex-equal-width ml2"
                    >
                      <div
                        class="medium ml4"
                      >
                        <div
                          class="enroll-acknowledgement-link"
                          data-qa="enroll-acknowledgement"
                        >
                          RULES_ACKNOWLEDGE_MESSAGE
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="blue inverted segment"
              color="blue"
            >
              <button
                class="continue button button--fluid huge button--yellow"
                data-qa="continue-button"
                type="submit"
              >
                ENROLL_CREATE_ACCOUNT
              </button>
            </div>
            <div
              class="segment"
            >
              <a
                class="pblue"
                href="/terms-and-conditions?clk=GNAVTC"
                target="_blank"
              >
                SHARED__FOOTER__TERMS_AND_CONDITIONS
              </a>
            </div>
          </div>
        </fieldset>
      </form>
    </div>
  </div>
</div>
`;
